### ISSUE TITLE
Include the entire second in the "To" filter

### ISSUE THREAD

**Issue by timgaunt** [datalust/seq-tickets#1166]
The `Search FROM timestamp` and `Search TO timestamp` menu items are fantastic and really helpful however `Search TO timestamp` would be so much better if it wasn't truncated to the start of the second as it removes the log event you're going _to_ because the milliseconds don't get taken into account (and can't be passed to the date filters:

![image](https://user-images.githubusercontent.com/254564/110970443-aa702d00-8351-11eb-98c7-e27b1b9e44b0.png)

Note the top entry with `null`, if this is then filtered by `Search TO timestamp`:
![image](https://user-images.githubusercontent.com/254564/110970489-b825b280-8351-11eb-8416-8ee98cd27634.png)

Then it is filtered out because it was on `.961`
![image](https://user-images.githubusercontent.com/254564/110970532-c1af1a80-8351-11eb-8d6d-f5ae1e30d6c4.png)

It would be awesome if the date filters could handle ms as a lot can happen in a second in which case `Search FROM timestamp` should also be to the ms however as an interim fix it would be great if it used the ceiling of the current second i.e. `25.961` would become `26` instead of the current `25`.

It's only a minor thing but would save a filter change everytime I used the feature.

**Comment by nblumhardt** [comment]
Thanks for the suggestion, Tim - we'll take a look 👍

**Comment by CrossBound** [comment]
I agree with this one. I think it makes more sense logically to use "through" a second instead of "to" a second.

**Comment by nblumhardt** [comment]
Thanks for all the feedback!

The behavior here isn't exactly as it seems; what's happening is that if the event has a timestamp `25.961`, searching to its timestamp _does_ actually use the full value, including milliseconds.

The reason the event doesn't appear in the results is that ranges in Seq have an inclusive start, and exclusive end. Because of this, searching to an event's timestamp doesn't include the event itself. Although there are some advantages for making _Search TO timestamp_ behave inclusively (even just by bumping up the value by an additional millisecond), I'm not keen to rush a change through without some more thought (we're only a few weeks from releasing 2021.3).

The date picker boxes are a bit more questionable in this case, though: they round the selected value for display, which is misleading. I've worked through some changes: when the underlying timestamp has a nonzero milliseconds component, the date pickers in 2021.3 will now show it, and dates typed manually into the picker will accept millisecond values:

![image](https://user-images.githubusercontent.com/342712/136156067-027f92de-0c2b-4af6-ad25-e9577688e4f2.png)

Tracking issue is https://github.com/datalust/seq-tickets/issues/1336.

I'll move the current ticket out of the 2021.3 milestone and loop back to it at the next milestone. Hopefully the investigation here, and the change to display of millisecond precision, at least clears up some of the confusion.

Thanks again!

**Comment by nblumhardt** [comment]
We're still considering whether we can improve the default options, here, but with the fix of #1336 we don't think this one will make the cut for 2022.1 or .2, so we'll continue the thread as a discussion for now.
